### PR TITLE
MUL-857: Checking if TX inputs are spendable.

### PR DESCRIPTION
**Pull request by Enmk**
Added check that input can be spen using given private key;
Added tests to verify;
Added power_split() function;
Added tetst for power_tests().

### CODE REVIEW

File: multy_core/src/bitcoin/bitcoin_account.cpp [hunk 1]
```diff
@@ -238,6 +238,20 @@ BitcoinAccount::~BitcoinAccount()
 {
 }
 
+void bitcoin_hash_160(const BinaryData& input, BinaryData* output)
```
Comment (PashaKlybik) [inline multy_core/src/bitcoin/bitcoin_account.cpp:241]: move this function to utility.cpp, it is needed in other chain

File: multy_core/src/bitcoin/bitcoin_transaction.cpp [hunk 2]
```diff
@@ -311,27 +347,13 @@ class BitcoinTransactionDestination
...
                   }),
-          sig_script()
+          sig_script(),
+          m_net_type(net_type)
     {
     }
 
     void update()
     {
-        BitcoinDataStream sig_stream;
```
Comment (PashaKlybik) [inline multy_core/src/bitcoin/bitcoin_transaction.cpp]: remove method ```BitcoinTransactionDestination.update``` and ```BitcoinTransactionSource.update```

**Review by PashaKlybik — CHANGES_REQUESTED** [review]
Please fix its.

File: multy_core/src/bitcoin/bitcoin_account.cpp (hunk 1 above)
Comment (Enmk) [reply multy_core/src/bitcoin/bitcoin_account.cpp:241]: It is not used anywhere else.

**Review by PashaKlybik — APPROVED** [review]
Ok

**Pull request merged**